examples/alarm/alarm.py

```diff
@@ -0,0 +1,83 @@
+import uuid
+from datetime import datetime
+
+from nisystemlink.clients.alarm import AlarmClient
+from nisystemlink.clients.alarm.models import (
+    AlarmOrderBy,
+    AlarmTransitionType,
+    CreateAlarmTransition,
+    CreateOrUpdateAlarmRequest,
+    QueryAlarmsWithFilterRequest,
+    TransitionInclusionOption,
+)
+from nisystemlink.clients.core import HttpConfiguration
+
+# Setup the server configuration to point to your instance of SystemLink Enterprise
+server_configuration = HttpConfiguration(
+    server_uri="https://yourserver.yourcompany.com",
+    api_key="YourAPIKeyGeneratedFromSystemLink",
+)
+client = AlarmClient(configuration=server_configuration)
+
+# Create a unique alarm ID for this example
+# alarm_id is a user-defined identifier for this alarm
+alarm_id = f"example_alarm_{uuid.uuid1().hex}"
+
+# Create an alarm with a SET transition
+create_request = CreateOrUpdateAlarmRequest(
+    alarm_id=alarm_id,
+    transition=CreateAlarmTransition(
+        transition_type=AlarmTransitionType.SET,
+        occurred_at=datetime.now(),
+        severity_level=3,
+        condition="Temperature exceeded threshold",
+        message="Temperature sensor reading: 85°C",
+    ),
+)
+# Returns instance_id - a server-generated unique identifier for this specific alarm occurrence
+id = client.create_or_update_alarm(create_request)
+
+# Get the alarm by its instance ID (the unique occurrence identifier)
+alarm = client.get_alarm(id)
+
+# Update the alarm with a higher severity (same alarm_id, updates the same instance)
+update_request = CreateOrUpdateAlarmRequest(
+    alarm_id=alarm_id,
+    transition=CreateAlarmTransition(
+        transition_type=AlarmTransitionType.SET,
+        occurred_at=datetime.now(),
+        severity_level=5,
+        condition="Temperature critically high",
+        message="Temperature sensor reading: 95°C",
+    ),
+)
+client.create_or_update_alarm(update_request)
+
+# Query alarms with a filter (can filter by alarm_id to find all instances)
+query_request = QueryAlarmsWithFilterRequest(
+    filter=f'alarmId="{alarm_id}"',
+    transition_inclusion_option=TransitionInclusionOption.ALL,
+    order_by=AlarmOrderBy.UPDATED_AT,
+    order_by_descending=True,
+    return_count=True,
+)
+query_response = client.query_alarms(query_request)
+
+# Acknowledge all queried alarms
+queried_alarm_ids = [alarm.instance_id for alarm in query_response.alarms]
```

Comment:
These are instance IDs, not alarm IDs (so, `queried_instance_ids`). However, it doesn't make sense to acknowledge more than one instance of the same alarm, because at most one instance can be active. I suppose with this query, it could return multiple active alarms if there are instances with the same alarm ID in different workspaces, but I wonder if we should just have it do `acknowledge_alarms(ids=[id])` and not tie it to the query response.